CRM/Core/BAO/ActionSchedule.php

```diff
@@ -1315,6 +1315,7 @@ public static function buildRecipientContacts($mappingID, $now, $params = array(
         }
         $insertAdditionalSql = "
 INSERT INTO civicrm_action_log (contact_id, entity_id, entity_table, action_schedule_id)
+SELECT * FROM (
```

Comment:
without this extra select the group-by get's undone when it comes to inserting. Not sure why but the tests demonstrate it as the current behaviour